sky/data/storage_utils.py

```diff
@@ -63,6 +66,43 @@ def format_storage_table(storages: List[Dict[str, Any]],
         return 'No existing storage.'
 
 
+def get_excluded_files_from_skyignore(src_dir_path: str) -> List[str]:
+    """List files and patterns ignored by the .skyignore file
+    in the given source directory.
+    """
+    excluded_list: List[str] = []
+    expand_src_dir_path = os.path.expanduser(src_dir_path)
+    skyignore_path = os.path.join(expand_src_dir_path,
+                                  constants.SKY_IGNORE_FILE)
+
+    try:
+        with open(skyignore_path, 'r', encoding='utf-8') as f:
+            for line in f:
+                line = line.strip()
+                if line and not line.startswith('#'):
+                    if '*' in line:
+                        # Make parsing consistent with rsync.
+                        if line.startswith('*.'):
+                            line = '**/' + line
```

Comment:
Just wondering if this will be too specific, e.g., does `test.log` match `test.log` in all subdir; or, `test*.log` match all sub dir? Wondering if we should add `**/` to all lines, except the ones start with `/`
(the newly proposed version might be very slow in a large folder with many files/folders)
```suggestion
                        if line.startswith('*.'):
                            line = '**/' + line
```

Reply:
No `test.log` and `test*.log` should be just matching in current dir. That being said, I should remove this logic and requires users to do `**/*` if they want to match ALL directories. I also added in doc saying users should avoid doing `*.txt` or `./*.txt`